subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/ResolveState.java

```diff
@@ -179,7 +179,8 @@ public NodeState getNode(ComponentState module, ConfigurationMetadata configurat
     }
 
     public SelectorState getSelector(DependencyState dependencyState, boolean ignoreVersion) {
-        SelectorState selectorState = selectors.computeIfAbsent(Pair.of(dependencyState.getRequested(), ignoreVersion), req -> {
+        boolean isVirtualPlatformEdge = dependencyState.getDependency() instanceof LenientPlatformDependencyMetadata;
+        SelectorState selectorState = selectors.computeIfAbsent(Pair.of(dependencyState.getRequested(), Pair.of(ignoreVersion, isVirtualPlatformEdge)), req -> {
```

Comment:
:o: Would it make sense to use an adhoc cache key now, rather than a couple of pairs?

Reply:
Yes, that makes total sense. Will rework that.